File: src/main/scala/com/ipponusa/strings/SimpleStringProducer.scala
package com.ipponusa.strings

import java.util.Properties

import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord}

object SimpleStringProducer extends App {

  val props = new Properties()
  props.put("bootstrap.servers", "localhost:9093")
  props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer")
  props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer")

  val topic = "mytopic"

  val producer = new KafkaProducer[String, String](props)
  for (i <- 0 to 1000) {
    val record = new ProducerRecord[String, String](topic, "value-" + i)
    producer.send(record)
    Thread.sleep(250)
  }

  producer.close()

}

File: kafka-client-kotlin/src/main/kotlin/com/seigneurin/strings/SimpleStringProducer.kt
package com.seigneurin.strings

import org.apache.kafka.clients.producer.KafkaProducer
import org.apache.kafka.clients.producer.ProducerRecord
import org.apache.kafka.common.serialization.StringSerializer

import java.util.Properties

fun main(args: Array<String>) {
    val props = Properties()
    props["bootstrap.servers"] = "localhost:9092"
    props["key.serializer"] = StringSerializer::class.java.canonicalName
    props["value.serializer"] = StringSerializer::class.java.canonicalName

    val producer = KafkaProducer<String, String>(props)
    for (i in 1..1000) {
        val record = ProducerRecord<String, String>("mytopic", "value-$i")
        producer.send(record)
        Thread.sleep(250)
    }

    producer.close()
}
